SimGeneral/TrackingAnalysis/plugins/SimHitTPAssociationProducer.cc

```diff
@@ -44,9 +48,9 @@ void SimHitTPAssociationProducer::produce(edm::Event& iEvent, const edm::EventSe
   }
 
   // PSimHits
-  for (auto psit=_simHitSrc.begin();psit<_simHitSrc.end();++psit) {
+  for (auto const& psit : _simHitSrc ) {
     edm::Handle<edm::PSimHitContainer>  PSimHitCollectionH;
-    iEvent.getByLabel(*psit,  PSimHitCollectionH);
+    iEvent.getByLabel(psit,  PSimHitCollectionH);
```

Comment:
Forgive my ignorance, but aren't we supposed to switch to getByToken in addition to declaring the consumes in the constructor?